source/api/commands/task.md

```diff
@@ -0,0 +1,177 @@
+
+{% note warning 'Anti-Pattern' %}
+Don't try to start a web server from `cy.task()`.
+
+Read about {% url 'best practices' best-practices#Web-Servers %} here.
+{% endnote %}
+
+# Syntax
+
+```javascript
+cy.task(event)
+cy.task(event, arg)
+cy.task(event, arg, options)
+```
+
+## Usage
+
+**{% fa fa-check-circle green %} Correct Usage**
+
+```javascript
+// in test
+cy.task('log', 'This will be output to the terminal')
+
+// in plugins file
+on('task:requested' (event, arg) => {
+  switch (event) {
+    case 'log':
+      console.log(arg)
+      return null
+  }
+})
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} event** ***(String)***
+
+An event name (can be any arbitrary string) that you handle via the `task:requested` event in the {% url "`pluginsFile`" configuration#Folders-Files %}.
+
+**{% fa fa-angle-right %} arg** ***(Object)***
+
+An argument to send along with the event. Can be any type of value that can be serialized by [JSON.stringify()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)  (string, boolean, array, object, etc.). Types such as functions, regular expressions, Symbols will be omitted or censored to `null`.
+
+**{% fa fa-angle-right %} options** ***(Object)***
+
+Pass in an options object to change the default behavior of `cy.task()`.
+
+Option | Default | Description
+--- | --- | ---
+`log` | `true` | {% usage_options log %}
+`timeout` | {% url `taskTimeout` configuration#Timeouts %} | {% usage_options timeout cy.task %}
+
+## Yields {% helper_icon yields %}
+
+`cy.task()` yields the value returned or resolved by the `task:requested` event in the {% url "`pluginsFile`" configuration#Folders-Files %}.
+
+# Examples
+
+## Command
+
+`cy.task()` provides an escape hatch for running arbitrary node code, so you can take actions necessary for your test outside the scope of Cypress. This is great for:
+
+- Seeding your test database
+- Storing state in node that you want persisted between tests
+- Performing parallel tasks (like making multiple http requests outside of Cypress)
+- Running an external process
+
+In the `task:requested` plugin event, the command will fail if `undefined` is returned. This helps catch typos or cases where the task event is not handled. If you don't need to return a value, explicitly return `null` to signal that the given event has been handled.
+
+***Log a string to the terminal***
+
+```javascript
+// in test
+cy.task('log', 'This will be output to the terminal')
+
+// in plugins file
+on('task:requested' (event, arg) => {
+  switch (event) {
+    case 'log':
+      console.log(arg) // This will log to the terminal
+      return null // Signal that the 'log' event has been handled
+  }
+})
+```
+
+***Read a JSON file's contents***
+
+Note: this serves as a demonstration only. We recommend using fixtures or {% url "`cy.readFile`" readfile %} for a more robust implementation of reading a file with Cypress.
```

Comment:
I don't think we should show an example that we don't recommend using. We should probably come up with some more solid examples...which takes more time unfortunately.

Reply:
Fair enough. I just didn't want some hand-wavy example with some pseudo-code for creating a user in a database or else some super complicated example that actually creates a user in a database.